index.js

```diff
@@ -11,96 +11,185 @@ const Type = (exports.Type = {
+        case TYPE.CONE: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createConeShape(hx, cylinderAxis);
+          break;
+        }
+        case TYPE.SPHERE: {
+          const radius =
+            fit === FIT.MANUAL && !isNaN(options.sphereRadius) ? options.sphereRadius : computeRadius(root);
+          collisionShape = new Ammo.btSphereShape(radius);
+          collisionShape.sphereRadius = radius;
+          break;
+        }
+        case TYPE.HULL: {
+          const hx = fit === FIT.MANUAL ? options.halfExtents : computeHalfExtents(root);
+          collisionShape = _createHullShape(root, fit, margin, bounds, options.hullMaxVertices || 100000);
+          break;
+        }
+        case TYPE.MESH: {
+          collisionShape = _createTriMeshShape(root, fit);
+          localOffset.add(pos);
+          break;
+        }
+        default:
+          console.warn(type + " is not currently supported");
+          return null;
       }
-      default:
-        console.warn(type + " is not currently supported");
-        return null;
-    }
+      //TODO: Support convex hull decomposition, compound shapes, gimpact (dynamic trimesh)
 
-    collisionShape.type = type;
-    collisionShape.setMargin(margin);
-    collisionShape.destroy = () => {
-      for (let res of collisionShape.resources || []) {
-        Ammo.destroy(res);
+      collisionShape.type = type;
+      collisionShape.setMargin(margin);
+      collisionShape.destroy = () => {
+        for (let res of collisionShape.resources || []) {
+          Ammo.destroy(res);
+        }
+        Ammo.destroy(collisionShape);
+      };
+
+      const localTransform = new Ammo.btTransform();
+      const rotation = new Ammo.btQuaternion();
+      localTransform.setIdentity();
+
+      if (fit === FIT.COMPOUND) {
+        localOffset.add(offset);
+        quat.multiply(orientation);
+      } else {
+        localOffset.copy(offset);
+        quat.copy(orientation);
       }
-      Ammo.destroy(collisionShape);
+
+      localTransform.getOrigin().setValue(localOffset.x, localOffset.y, localOffset.z);
+      rotation.setValue(quat.x, quat.y, quat.z, quat.w);
+
+      localTransform.setRotation(rotation);
+      Ammo.destroy(rotation);
+
+      const localScale = new Ammo.btVector3(scale.x, scale.y, scale.z);
+      collisionShape.setLocalScaling(localScale);
+      Ammo.destroy(localScale);
+
+      collisionShape.localTransform = localTransform;
+
+      return collisionShape;
     };
 
-    return collisionShape;
+    const shapes = [];
+    matrix.identity();
+    if (fit === FIT.COMPOUND) {
+      inverse.getInverse(sceneRoot.matrixWorld);
+      sceneRoot.traverse(obj => {
+        if (obj.isMesh && (!THREE.Sky || obj.__proto__ != THREE.Sky.prototype)) {
+          matrix.multiplyMatrices(inverse, obj.matrixWorld);
+          matrix.decompose(pos, quat, scale);
+          shapes.push(createCollisionShape(obj));
```

Comment:
It seems confusing to me that this "sets up" `pos`, `quat`, and `scale` to be implicitly used by `createCollisionShape` above. Functions should take parameters.